File: NoteS/models/entity/Account.cs
﻿using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace NoteS.models.entity;

[Table("accounts")]
public partial class Account(string name, string uuid)
{
    [Column("id")] public int? Id { get; set; }

    [Column("name")] [MaxLength(128)] public string Name { get; set; } = name;

    [Column("uuid")] [MaxLength(128)] public string Uuid { get; set; } = uuid;

}

File: javaBack/ResourceServer/src/main/java/com/notes/models/Account.java
package com.notes.models;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.*;
import org.hibernate.validator.constraints.Length;

import java.time.Instant;

import static jakarta.persistence.GenerationType.SEQUENCE;
import static lombok.AccessLevel.PROTECTED;

@Entity
@Table(name = "accounts")
@NoArgsConstructor(access = PROTECTED)
@AllArgsConstructor(access = PROTECTED)
@Builder
@Getter
public class Account {
    @Id
    @GeneratedValue(strategy = SEQUENCE, generator = "account_seq")
    @SequenceGenerator(name = "account_seq", sequenceName = "account_id_seq")
    Long id;

    @NotNull
    @NotEmpty
    @Length(max = 128)
    @Column(name = "name", nullable = false, unique = true)
    String name;

    @NotNull
    @NotEmpty
    @Length(max = 128)
    @Column(name = "uuid", nullable = false, unique = true)
    String uuid;
}